src/core/SpriteAnimator.tsx

```diff
@@ -1,98 +1,177 @@
   maxItems?: number
-  instanceItems?: any[]
-  spriteDataset?: any
+  /** Pre-parsed sprite data, usually from useSpriteLoader ready for use */
+  spriteDataset?: {
+    spriteTexture: THREE.Texture
+    spriteData: SpriteData
+  }
+  /** Configuration options for the canvas context when loading textures
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/CanvasRenderingContext2D
+   */
   canvasRenderingContext2DSettings?: CanvasRenderingContext2DSettings
+  /** Controls whether frame positions are rounded for precise pixel alignment.
+   * Enable this if you notice slight texture bleeding between frames.
+   */
   roundFramePosition?: boolean
-  /** Additional properties to be passed to the underlying mesh/instance
+  /** Additional properties to be passed to both simple mesh and instance components.
    * @example { frustumCulled: false, renderOrder: 1 }
+   * @see https://threejs.org/docs/#api/en/core/Object3D
    */
   meshProps?: CommonMeshProps
 } & JSX.IntrinsicElements['group']
 
 type SpriteAnimatorState = {
-  /** The user-defined, mutable, current goal position along the curve, it may be >1 or <0 */
-  current: number | undefined
-  /** The 0-1 normalised and damped current goal position along curve */
-  offset: number | undefined
-  hasEnded: boolean | undefined
-  ref: React.MutableRefObject<any> | undefined | null | ((instance: any) => void)
+  current?: number
+  offset?: number
+  imageUrl?: string
+  hasEnded: boolean
+  ref: React.RefObject<THREE.Group> | null | ((instance: THREE.Group) => void)
 }
 
-type Scale = number | [number, number, number] | Vector3
+type Scale = Vector3
 
 const context = React.createContext<SpriteAnimatorState | null>(null)
 
 export function useSpriteAnimator() {
   return React.useContext(context)
 }
 
+// Type guard for SpriteData
+function isSpriteData(data: SpriteData | null) {
+  return data !== null && 'meta' in data && 'frames' in data
+}
+
 const geometry = new THREE.PlaneGeometry(1, 1)
 
-export const SpriteAnimator: React.FC<SpriteAnimatorProps> = /* @__PURE__ */ React.forwardRef(
+export const SpriteAnimator = /* @__PURE__ */ React.forwardRef<THREE.Group, SpriteAnimatorProps>(
   (
     {
-      startFrame,
+      startFrame = 0,
       endFrame,
-      fps,
-      frameName,
+      fps = 30,
+      frameName = '',
       textureDataURL,
       textureImageURL,
-      loop,
+      loop = false,
       numberOfFrames,
-      autoPlay,
+      autoPlay = true,
       animationNames,
       onStart,
       onEnd,
       onLoopEnd,
       onFrame,
       play,
-      pause,
-      flipX,
-      alphaTest,
+      pause = false,
+      flipX = false,
+      alphaTest = 0.0,
       children,
-      asSprite,
+      asSprite = false,
       offset,
       playBackwards,
       resetOnEnd,
-      maxItems,
+      maxItems = 1,
       instanceItems,
       spriteDataset,
       canvasRenderingContext2DSettings,
```

Comment:
you should set as many default as possible since it will reduce the complexity of handling with `undefined` values into the code